semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -427,7 +427,10 @@ let m_regexp_options a_opt b_opt =
 (* TODO: factorize with metavariable and aliasing logic in m_expr
  * TODO: remove MV.Id and use always MV.N?
  *)
-let rec m_name a b =
+(* `is_resolved` should be true if `b` is the result of unpacking a `ResolvedName`.
+   See the case conditioning on `is_resolved` below.
+*)
+let rec m_name ?(is_resolved = false) a b =
   let try_parents dotted =
```

Comment:
I wonder if this `hook_find_possible_parents` may also be a source of ids with mixed tokens. These will be the parent classes of `dotted` (assuming `dotted` represents a class).